typescript/sdk/src/token/EvmERC20WarpRouteReader.ts

```diff
@@ -111,15 +181,45 @@ export class EvmERC20WarpRouteReader {
    * @param tokenAddress - The address of the token.
    * @returns A partial ERC20 metadata object containing the token name, symbol, total supply, and decimals.
    */
-  async fetchTokenMetadata(tokenAddress: Address): Promise<ERC20Metadata> {
-    const erc20 = ERC20__factory.connect(tokenAddress, this.provider);
-    const [name, symbol, totalSupply, decimals] = await Promise.all([
+  async fetchTokenMetadata(
+    type: DerivedTokenType,
+    tokenAddress: Address,
+  ): Promise<ERC20Metadata & { token?: string }> {
+    if (type === TokenType.collateral || type === TokenType.collateralVault) {
+      const erc20 = HypERC20Collateral__factory.connect(
+        tokenAddress,
+        this.provider,
+      );
+      const token = await erc20.wrappedToken();
+      const { name, symbol, decimals, totalSupply } =
+        await this.fetchERC20Metadata(token);
+
+      return { name, symbol, decimals, totalSupply, token };
+    } else if (type === TokenType.synthetic) {
+      return this.fetchERC20Metadata(tokenAddress);
+    } else {
+      // Assumes Native
```

Comment:
I think this is a bad assumption
eg I just added some new token types which would fall under "collateral"

Reply:
fixed.